packages/instantsearch.js/src/lib/InstantSearch.ts

```diff
@@ -297,12 +298,17 @@ See ${createDocumentationLink({
     this.sendEventToInsights = noop;
 
     if (routing) {
-      const routerOptions = typeof routing === 'boolean' ? undefined : routing;
+      const routerOptions = typeof routing === 'boolean' ? {} : routing;
+      routerOptions.$$internal = true;
       this.use(createRouterMiddleware(routerOptions));
     }
 
+    // This is the default middleware,
+    // any user-provided middleware will be added later and override this one.
+    this.use(createInsightsMiddleware({ $$internal: true }));
```

Comment:
something still needs to be fixed here, the script is always loaded from the internal middleware, even if users add a different middleware later (even right after). Not too sure how to fix that yet

Reply:
solved through https://github.com/algolia/instantsearch/pull/5493